api/admin/controller/individual_admin_settings.py

```diff
@@ -98,6 +98,104 @@ def append_role(roles, role):
             individualAdmins=admins,
         )
 
+    def process_post_create_first_admin(self, email: str):
+        """Create the first admin in the system."""
+
+        # Passwords are always required, so check presence and validity up front.
+        password: Optional[str] = flask.request.form.get("password")
+        if not self.is_acceptable_password(password):
+            return self.unacceptable_password()
+
+        success = False
+        try:
+            admin, _ = get_one_or_create(self._db, Admin, email=email)
+            self.check_permissions(admin, settingUp=True)
+
+            # Update the roles, if requested.
+            roles = flask.request.form.get("roles")
+            if roles:
+                roles = json.loads(roles)
+            else:
+                roles = []
+
+            roles_error = self.handle_roles(admin, roles, settingUp=True)
+            if roles_error:
+                return roles_error
+
+            # Update the password, if requested.
+            self.handle_password(password, admin, is_new=True, settingUp=True)
+
+            success = True
+            return self.response(admin, is_new=True)
+        finally:
+            if not success:
+                self._db.rollback()
+
```

Comment:
This is the main issue that was being fixed; previously, the code could "fail" and yet _still_ create an admin with an empty password.